llmfoundry/__init__.py

```diff
@@ -24,7 +24,8 @@
 
 except ImportError as e:
     try:
-        is_cuda_available = torch.cuda.is_available()  # type: ignore
+        import torch
```

Comment:
Should we import torch before the large try block above?

Reply:
I think the intention here was to raise a nice error anytime someone did not have the necessary reqs (including torch). We've now ended up in a situation where you have to install torch _before_ installing the repo, so this try/except for the torch import and follow on message may not be the most helpful way of doing it. we can probably just move the `import torch` out of the try/except as you said.